core/appeal/service_test.go

```diff
@@ -697,18 +717,53 @@ func (s *ServiceTestSuite) TestCreate() {
 				},
 				AppealConfig: &domain.PolicyAppealConfig{AllowOnBehalf: true},
 			},
+			{
+				ID:      "policy_2",
+				Version: 1,
+				Steps: []*domain.Step{
+					{
+						Name:     "step_1",
+						Strategy: "manual",
+						Approvers: []string{
+							"$appeal.resource.details.owner",
+						},
+					},
+					{
+						Name:     "step_2",
+						Strategy: "manual",
+						Approvers: []string{
+							`$appeal.creator != nil ? $appeal.creator.managers : "approver@example.com"`,
+						},
+					},
+				},
+				IAM: &domain.IAMConfig{
+					Provider: "http",
+					Config: map[string]interface{}{
+						"url": "http://localhost",
+					},
+					Schema: map[string]string{
+						"managers": `managers`,
+						"name":     "name",
+						"role":     `$response.roles[0].name`,
+						"roles":    `map($response.roles, {#.name})`,
+					},
+				},
+				AppealConfig: &domain.PolicyAppealConfig{
+					AllowOnBehalf:  true,
+					IgnoreIAMError: true,
```

Comment:
Assuming a policy step approvers like this -  `$appeal.creator != nil ? $appeal.creator.managers : "approver@example.com"`

What happens if there is an error in the IAM provider and there is no response. Since these changes allow the appeal to be created, who would be the approver for this step of the appeal?

Reply:
as expected in this test itself, the expression will evaluate to `approver@example.com` since the appeal.creator is nil https://github.com/goto/guardian/pull/33/files#diff-19236a37354d81d4c7688919234a3a2bfe9579b81f8ff5ba6db2dee24370bc00R895

Reply:
But that is not error checking right? There is no IAM error here. With/without the IAMError flag - the same would have happened.

Reply:
error checking is on [this line](https://github.com/goto/guardian/pull/33/files#diff-4ceaf11357ea8ac162487cfbb8f9f0a0a7446fa7b61bc799aac1a3d3ca7adc58R1076-R1079) (error testing mock is [in here](https://github.com/goto/guardian/pull/33/files#diff-19236a37354d81d4c7688919234a3a2bfe9579b81f8ff5ba6db2dee24370bc00R931)). The error from iam service was ignored by enabling `ignore_iam_error` and then the appeal creation continues with `appeal.creator = nil` (previously, appeal creation returns error at this point) therefore the evaluation value is based on the else condition of the expression